x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_params_schema.ts

```diff
@@ -1,86 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-import { schema, TypeOf } from '@kbn/config-schema';
-
-import { DEFAULT_MAX_SIGNALS } from '../../../../common/constants';
-
-export const signalSchema = schema.object({
-  anomalyThreshold: schema.maybe(schema.number()),
-  author: schema.arrayOf(schema.string(), { defaultValue: [] }),
-  buildingBlockType: schema.nullable(schema.string()),
-  description: schema.string(),
-  note: schema.nullable(schema.string()),
-  eventCategoryOverride: schema.maybe(schema.string()),
-  falsePositives: schema.arrayOf(schema.string(), { defaultValue: [] }),
-  from: schema.string(),
-  ruleId: schema.string(),
-  immutable: schema.boolean({ defaultValue: false }),
-  index: schema.nullable(schema.arrayOf(schema.string())),
-  language: schema.nullable(schema.string()),
-  license: schema.nullable(schema.string()),
-  outputIndex: schema.nullable(schema.string()),
-  savedId: schema.nullable(schema.string()),
-  timelineId: schema.nullable(schema.string()),
-  timelineTitle: schema.nullable(schema.string()),
-  meta: schema.nullable(schema.object({}, { unknowns: 'allow' })),
-  machineLearningJobId: schema.maybe(schema.string()),
-  query: schema.nullable(schema.string()),
-  filters: schema.nullable(schema.arrayOf(schema.object({}, { unknowns: 'allow' }))),
-  maxSignals: schema.number({ defaultValue: DEFAULT_MAX_SIGNALS }),
-  riskScore: schema.number(),
-  // TODO: Specify types explicitly since they're known?
-  riskScoreMapping: schema.nullable(schema.arrayOf(schema.object({}, { unknowns: 'allow' }))),
-  ruleNameOverride: schema.nullable(schema.string()),
-  severity: schema.string(),
-  severityMapping: schema.nullable(schema.arrayOf(schema.object({}, { unknowns: 'allow' }))),
-  threat: schema.nullable(schema.arrayOf(schema.object({}, { unknowns: 'allow' }))),
-  threshold: schema.maybe(
-    schema.object({
-      // Can be an empty string (pre-7.12) or empty array (7.12+)
-      field: schema.nullable(
-        schema.oneOf([schema.string(), schema.arrayOf(schema.string(), { maxSize: 3 })])
```

Comment:
For posterity's sake, want to record that these `maxSize` restrictions will be added back in a future pr